diff --git a/src/audioscrobbler_api.h b/src/audioscrobbler_api.h
index 56b8f3c..fc13683 100644
--- a/src/audioscrobbler_api.h
+++ b/src/audioscrobbler_api.h
@@ -261,11 +261,11 @@ static void api_get_signature(const char *string, const char *secret, char *resu
     const size_t string_len = strlen(string);
     const size_t secret_len = strlen(secret);
     const size_t len = string_len + secret_len;
-    char sig[MAX_PROPERTY_LENGTH+1] = {0};
+    char sig[2*MAX_PROPERTY_LENGTH+1] = {0};
 
     // TODO(marius): this needs to change to memcpy or strncpy
-    strncat(sig, string, MAX_PROPERTY_LENGTH/2);
-    strncat(sig, secret, MAX_PROPERTY_LENGTH/2);
+    strncat(sig, string, MAX_PROPERTY_LENGTH);
+    strncat(sig, secret, MAX_PROPERTY_LENGTH);
 
     unsigned char sig_hash[MD5_DIGEST_LENGTH] = {0};
 
